### ISSUE TITLE
new rule - require comment explaining `type:ignore` comments

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#8454]
it would be nice to have a rule that enforces that `type:ignore` comments have a comment above them describing the reason the error is being suppressed, eg:

incorrect:
```py
def foo():
    a: str = 1  # type:ignore[assignment] # error: add a comment on the line above explaining why the error is being ignored
```

correct:
```py
def foo():
    # false positive, see https://github.com/python/mypy/issues/1
    a: str = 1  # type:ignore[assignment]
```

eslint has a rule for this: https://typescript-eslint.io/rules/ban-ts-comment/

related: #8453

**Comment by Avasam** [comment]
Does this request extend to suppression comments in general? (`# pyright: ignore`, `# noqa`, and other suppressions that ruff supports)
eslint equivalent: https://eslint-community.github.io/eslint-plugin-eslint-comments/rules/require-description.html

**Comment by DetachHead** [comment]
true, thanks. i updated the issue
